### PR TITLE
Add ServerMode client mutation coverage

### LINKED ISSUE #11212: [mutation-test-improver] Monthly Report 2026-09
## Suggested Actions for Maintainer

* [ ] **Review PR**: kills mutant in `src/Platform/Microsoft.Testing.Platform.ServerMode.Client.Sources/Client/MtpServerConnector.cs:253` - branch `mutation-test-improver/wait-bounded-async-task-completed` (PR creation was queued this run; the number will be added once reconciled).
* [ ] **Investigate survived hotspot** `MtpServerInProcessHost.cs`: 52 survived + 15 no-coverage mutants, none auto-fixed this run - [mutation report artifact](https://github.com/microsoft/testfx/actions/runs/34674810397)
* [ ] **Investigate survived hotspot** `MtpServerProcess.cs`: 31 survived + 39 no-coverage mutants, none auto-fixed this run
* [ ] **Investigate survived hotspot** `MtpServerClient.cs`: 27 survived + 35 no-coverage mutants, none auto-fixed this run
* [ ] **Investigate survived hotspot** `MtpJsonRpcConnection.cs`: 25 survived + 18 no-coverage mutants, none auto-fixed this run
* [ ] **Investigate survived hotspot** `SerializerUtilities.ClientSerializers.cs`: 6 survived + 27 no-coverage mutants (mostly the `RegisterClientSerializers` idempotent double-checked-lock and deserializer edge cases), none auto-fixed this run
* [ ] **Follow-up on `MtpServerConnector.WaitBoundedAsync`**: two equality-mutation mutants on the timeout-clamping comparisons (lines 256 and 261) remained `Survived` after this run's fix; worth a second pass to add boundary-value tests for `timeout == TimeSpan.Zero` and `timeout == MaxDelay` exactly.

## Score Trend

| Date | Score | Killed | Survived | No Coverage | Timeout | Δ vs previous |
| --- | --- | --- | --- | --- | --- | --- |
| 2026-09-12 | 37.35% | 140 | 166 | 141 | 43 | (first tracked run) |

*(Keep the most recent ~14 rows; drop older ones - full history remains in Run History below.)*

## Top Survived-Mutant Hotspots

Ranked by combined `Survived` + `NoCoverage` count (the actionable, undetected gaps). `CompileError`/`RuntimeError`/`Ignored` mutants are excluded from ranking and the score denominator.

1. **`Client/MtpServerInProcessHost.cs`** - Survived: 52, NoCoverage: 15
2. **`Client/MtpServerProcess.cs`** - Survived: 31, NoCoverage: 39
3. **`Client/MtpServerClient.cs`** - Survived: 27, NoCoverage: 35
4. **`Client/MtpJsonRpcConnection.cs`** - Survived: 25, NoCoverage: 18
5. **`Client/MtpServerConnector.cs`** - Survived: 25 (2 remain after this run's fix), NoCoverage: 7
6. **`Client/SerializerUtilities.ClientSerializers.cs`** - Survived: 6, NoCoverage: 27

Source report: [mutation-testing-report artifact from run 34674810397](https://github.com/microsoft/testfx/actions/runs/34674810397).

## Known Equivalent Mutants

*(None recorded yet. Candidates considered this run but not yet classified as equivalent - kept as open hotspots above pending deeper investigation: the `RegisterClientSerializers` double-checked-locking idempotency mutants (`SerializerUtilities.ClientSerializers.cs`) may be partially equivalent under single-threaded test execution but were not investigated deeply enough this run to record with confidence.)*

### 2026-09-12 06:46 UTC - [Run](https://github.com/microsoft/testfx/actions/runs/34674810397)
- Score: 37.35% (Killed 140 / Survived 166 / No Coverage 141 / Timeout 43)
- 🔧 Opened PR (creation queued): kills mutant in `Client/MtpServerConnector.cs:253` (`WaitBoundedAsync` task.IsCompleted boolean mutation) - branch `mutation-test-improver/wait-bounded-async-task-completed`. Verified locally: new tests pass against unmutated code, and a full local Stryker re-run confirmed mutant id 401 flipped from `Survived` to `Killed` (project score after fix: 37.55%, Killed 142 / Survived 164).
- 🟰 No other fix attempted this run: verification budget (2 total Stryker runs) was spent on this candidate; `MtpServerConnector.cs` timeout-clamping equality mutants (lines 256, 261) remained survived and are carried to Suggested Actions above.

> [!WARNING]
> <details>
> <summary>Firewall blocked 1 domain</summary>
>
> The following domain was blocked by the firewall during workflow execution:
>
> - `southcentralus0.in.applicationinsights.azure.com`
>
> To allow these domains, add them to the `network.allowed` list in your workflow frontmatter:
>
> ```yaml
> network:
>   allowed:
>     - defaults
>     - "southcentralus0.in.applicationinsights.azure.com"
> ```
>
> See [Network Configuration](https://github.github.com/gh-aw/reference/network/) for more information.
>
> </details>

> 🤖 **Automated content by GitHub Copilot.** Generated by the [Mutation Test Improver](https://github.com/microsoft/testfx/actions/runs/34676193328/agentic_workflow) workflow. · auto · 757.2 AIC · ⌖ 15.9 AIC · ⊞ 15.1K · [◷]( · [◷](https://github.com/search?q=repo%3Amicrosoft%2Ftestfx+is%3Aissue+%22gh-aw-workflow-call-id%3A+microsoft%2Ftestfx%2Fmutation-test-improver%22&type=issues))

---

## Suggested Actions for Maintainer

* [ ] **Pending PR creation** for `Client/MtpServerConnector.cs:253` (`WaitBoundedAsync` task.IsCompleted boolean mutation): verified fix was queued by safe outputs in the 2026-09-12 06:46 UTC run and will be reconciled on a future successful run.
* [ ] **Investigate survived hotspot** `MtpServerInProcessHost.cs`: 52 survived + 15 no-coverage mutants, none auto-fixed this run - [mutation report artifact](https://github.com/microsoft/testfx/actions/runs/34674810397)
* [ ] **Investigate survived hotspot** `MtpServerProcess.cs`: 31 survived + 39 no-coverage mutants, none auto-fixed this run
* [ ] **Investigate survived hotspot** `MtpServerClient.cs`: 27 survived + 35 no-coverage mutants, none auto-fixed this run
* [ ] **Investigate survived hotspot** `MtpJsonRpcConnection.cs`: 25 survived + 18 no-coverage mutants, none auto-fixed this run
* [ ] **Investigate survived hotspot** `SerializerUtilities.ClientSerializers.cs`: 6 survived + 27 no-coverage mutants (mostly the `RegisterClientSerializers` idempotent double-checked-lock and deserializer edge cases), none auto-fixed this run
* [ ] **Follow-up on `MtpServerConnector.WaitBoundedAsync`**: two equality-mutation mutants on the timeout-clamping comparisons (lines 256 and 261) remained `Survived` after the 2026-09-12 run's fix; worth a second pass to add boundary-value tests for `timeout == TimeSpan.Zero` and `timeout == MaxDelay` exactly.

## Score Trend

| Date | Score | Killed | Survived | No Coverage | Timeout | Δ vs previous |
| --- | --- | --- | --- | --- | --- | --- |
| 2026-09-12 | 37.35% | 140 | 166 | 141 | 43 | (first tracked run) |

*(Keep the most recent ~14 rows; drop older ones - full history remains in Run History below.)*

## Top Survived-Mutant Hotspots

Ranked by combined `Survived` + `NoCoverage` count (the actionable, undetected gaps). `CompileError`/`RuntimeError`/`Ignored` mutants are excluded from ranking and the score denominator.

1. **`Client/MtpServerInProcessHost.cs`** - Survived: 52, NoCoverage: 15
2. **`Client/MtpServerProcess.cs`** - Survived: 31, NoCoverage: 39
3. **`Client/MtpServerClient.cs`** - Survived: 27, NoCoverage: 35
4. **`Client/MtpJsonRpcConnection.cs`** - Survived: 25, NoCoverage: 18
5. **`Client/MtpServerConnector.cs`** - Survived: 25 (2 remain after the 2026-09-12 run's fix), NoCoverage: 7
6. **`Client/SerializerUtilities.ClientSerializers.cs`** - Survived: 6, NoCoverage: 27

Source report: [mutation-testing-report artifact from run 34674810397](https://github.com/microsoft/testfx/actions/runs/34674810397). No newer report is available yet — the 2026-09-13 run failed before producing one.

## Known Equivalent Mutants

*(None recorded yet. Candidates considered but not yet classified as equivalent - kept as open hotspots above pending deeper investigation: the `RegisterClientSerializers` double-checked-locking idempotency mutants (`SerializerUtilities.ClientSerializers.cs`) may be partially equivalent under single-threaded test execution but were not investigated deeply enough to record with confidence.)*

### 2026-09-13 05:18 UTC - [Run](https://github.com/microsoft/testfx/actions/runs/34739561186)
- ⚠️ Run failed/cancelled - no report available

### 2026-09-12 06:46 UTC - [Run](https://github.com/microsoft/testfx/actions/runs/34674810397)
- Score: 37.35% (Killed 140 / Survived 166 / No Coverage 141 / Timeout 43)
- 🔧 Opened PR (creation queued): kills mutant in `Client/MtpServerConnector.cs:253` (`WaitBoundedAsync` task.IsCompleted boolean mutation) - branch `mutation-test-improver/wait-bounded-async-task-completed`. Verified locally: new tests pass against unmutated code, and a full local Stryker re-run confirmed mutant id 401 flipped from `Survived` to `Killed` (project score after fix: 37.55%, Killed 142 / Survived 164).
- 🟰 No other fix attempted this run: verification budget (2 total Stryker runs) was spent on this candidate; `MtpServerConnector.cs` timeout-clamping equality mutants (lines 256, 261) remained survived and are carried to Suggested Actions above.> 🤖 **Automated content by GitHub Copilot.** Generated by the [Mutation Test Improver](https://github.com/microsoft/testfx/actions/runs/34739830516/agentic_workflow) workflow. · auto · 63.9 AIC · ⌖ 2.43 AIC · ⊞ 15.1K · [◷]( · [◷](https://github.com/search?q=repo%3Amicrosoft%2Ftestfx+is%3Aissue+%22gh-aw-workflow-call-id%3A+microsoft%2Ftestfx%2Fmutation-test-improver%22&type=issues))

---

## Suggested Actions for Maintainer

* [ ] **Pending PR creation** for `Client/MtpServerConnector.cs:253` (`WaitBoundedAsync` task.IsCompleted boolean mutation): verified fix was queued by safe outputs in the 2026-09-12 06:46 UTC run and has not yet materialized as an open PR (no matching branch/PR found on GitHub as of this run); will be retried/reconciled on a future successful run.
* [ ] **Investigate survived hotspot** `MtpServerInProcessHost.cs`: 52 survived + 15 no-coverage mutants, none auto-fixed this run - [mutation report artifact](https://github.com/microsoft/testfx/actions/runs/34674810397)
* [ ] **Investigate survived hotspot** `MtpServerProcess.cs`: 31 survived + 39 no-coverage mutants, none auto-fixed this run
* [ ] **Investigate survived hotspot** `MtpServerClient.cs`: 27 survived + 35 no-coverage mutants, none auto-fixed this run
* [ ] **Investigate survived hotspot** `MtpJsonRpcConnection.cs`: 25 survived + 18 no-coverage mutants, none auto-fixed this run
* [ ] **Investigate survived hotspot** `SerializerUtilities.ClientSerializers.cs`: 6 survived + 27 no-coverage mutants (mostly the `RegisterClientSerializers` idempotent double-checked-lock and deserializer edge cases), none auto-fixed this run
* [ ] **Follow-up on `MtpServerConnector.WaitBoundedAsync`**: two equality-mutation mutants on the timeout-clamping comparisons (lines 256 and 261) remained `Survived` after the 2026-09-12 run's fix; worth a second pass to add boundary-value tests for `timeout == TimeSpan.Zero` and `timeout == MaxDelay` exactly.

## Score Trend

| Date | Score | Killed | Survived | No Coverage | Timeout | Δ vs previous |
| --- | --- | --- | --- | --- | --- | --- |
| 2026-09-12 | 37.35% | 140 | 166 | 141 | 43 | (first tracked run) |

*(Keep the most recent ~14 rows; drop older ones - full history remains in Run History below.)*

## Top Survived-Mutant Hotspots

Ranked by combined `Survived` + `NoCoverage` count (the actionable, undetected gaps). `CompileError`/`RuntimeError`/`Ignored` mutants are excluded from ranking and the score denominator.

1. **`Client/MtpServerInProcessHost.cs`** - Survived: 52, NoCoverage: 15
2. **`Client/MtpServerProcess.cs`** - Survived: 31, NoCoverage: 39
3. **`Client/MtpServerClient.cs`** - Survived: 27, NoCoverage: 35
4. **`Client/MtpJsonRpcConnection.cs`** - Survived: 25, NoCoverage: 18
5. **`Client/MtpServerConnector.cs`** - Survived: 25 (2 remain after the 2026-09-12 run's fix), NoCoverage: 7
6. **`Client/SerializerUtilities.ClientSerializers.cs`** - Survived: 6, NoCoverage: 27

Source report: [mutation-testing-report artifact from run 34674810397](https://github.com/microsoft/testfx/actions/runs/34674810397). No newer report is available yet - the 2026-09-13 and 2026-09-14 runs both failed before producing one.

## Known Equivalent Mutants

*(None recorded yet. Candidates considered but not yet classified as equivalent - kept as open hotspots above pending deeper investigation: the `RegisterClientSerializers` double-checked-locking idempotency mutants (`SerializerUtilities.ClientSerializers.cs`) may be partially equivalent under single-threaded test execution but were not investigated deeply enough to record with confidence.)*

### 2026-09-14 05:15 UTC - [Run](https://github.com/microsoft/testfx/actions/runs/34808600446)
- ⚠️ Run failed/cancelled - no report available

### 2026-09-13 05:18 UTC - [Run](https://github.com/microsoft/testfx/actions/runs/34739561186)
- ⚠️ Run failed/cancelled - no report available

### 2026-09-12 06:46 UTC - [Run](https://github.com/microsoft/testfx/actions/runs/34674810397)
- Score: 37.35% (Killed 140 / Survived 166 / No Coverage 141 / Timeout 43)
- 🔧 Opened PR (creation queued): kills mutant in `Client/MtpServerConnector.cs:253` (`WaitBoundedAsync` task.IsCompleted boolean mutation) - branch `mutation-test-improver/wait-bounded-async-task-completed`. Verified locally: new tests pass against unmutated code, and a full local Stryker re-run confirmed mutant id 401 flipped from `Survived` to `Killed` (project score after fix: 37.55%, Killed 142 / Survived 164).
- 🟰 No other fix attempted this run: verification budget (2 total Stryker runs) was spent on this candidate; `MtpServerConnector.cs` timeout-clamping equality mutants (lines 256, 261) remained survived and are carried to Suggested Actions above.> 🤖 **Automated content by GitHub Copilot.** Generated by the [Mutation Test Improver](https://github.com/microsoft/testfx/actions/runs/34808988647/agentic_workflow) workflow. · auto · 59 AIC · ⌖ 2.34 AIC · ⊞ 15.1K · [◷]( · [◷](https://github.com/search?q=repo%3Amicrosoft%2Ftestfx+is%3Aissue+%22gh-aw-workflow-call-id%3A+microsoft%2Ftestfx%2Fmutation-test-improver%22&type=issues))

---

## Suggested Actions for Maintainer

* [ ] **Pending PR creation** for `Client/MtpServerConnector.cs:253` (`WaitBoundedAsync` task.IsCompleted boolean mutation): verified fix was queued by safe outputs in the 2026-09-12 06:46 UTC run and has not yet materialized as an open PR (no matching branch/PR found on GitHub as of this run); will be retried/reconciled on a future successful run.
* [ ] **Investigate survived hotspot** `MtpServerInProcessHost.cs`: 52 survived + 15 no-coverage mutants, none auto-fixed this run - [mutation report artifact](https://github.com/microsoft/testfx/actions/runs/34674810397)
* [ ] **Investigate survived hotspot** `MtpServerProcess.cs`: 31 survived + 39 no-coverage mutants, none auto-fixed this run
* [ ] **Investigate survived hotspot** `MtpServerClient.cs`: 27 survived + 35 no-coverage mutants, none auto-fixed this run
* [ ] **Investigate survived hotspot** `MtpJsonRpcConnection.cs`: 25 survived + 18 no-coverage mutants, none auto-fixed this run
* [ ] **Investigate survived hotspot** `SerializerUtilities.ClientSerializers.cs`: 6 survived + 27 no-coverage mutants (mostly the `RegisterClientSerializers` idempotent double-checked-lock and deserializer edge cases), none auto-fixed this run
* [ ] **Follow-up on `MtpServerConnector.WaitBoundedAsync`**: two equality-mutation mutants on the timeout-clamping comparisons (lines 256 and 261) remained `Survived` after the 2026-09-12 run's fix; worth a second pass to add boundary-value tests for `timeout == TimeSpan.Zero` and `timeout == MaxDelay` exactly.

## Score Trend

| Date | Score | Killed | Survived | No Coverage | Timeout | Δ vs previous |
| --- | --- | --- | --- | --- | --- | --- |
| 2026-09-12 | 37.35% | 140 | 166 | 141 | 43 | (first tracked run) |

*(Keep the most recent ~14 rows; drop older ones - full history remains in Run History below.)*

## Top Survived-Mutant Hotspots

Ranked by combined `Survived` + `NoCoverage` count (the actionable, undetected gaps). `CompileError`/`RuntimeError`/`Ignored` mutants are excluded from ranking and the score denominator.

1. **`Client/MtpServerInProcessHost.cs`** - Survived: 52, NoCoverage: 15
2. **`Client/MtpServerProcess.cs`** - Survived: 31, NoCoverage: 39
3. **`Client/MtpServerClient.cs`** - Survived: 27, NoCoverage: 35
4. **`Client/MtpJsonRpcConnection.cs`** - Survived: 25, NoCoverage: 18
5. **`Client/MtpServerConnector.cs`** - Survived: 25 (2 remain after the 2026-09-12 run's fix), NoCoverage: 7
6. **`Client/SerializerUtilities.ClientSerializers.cs`** - Survived: 6, NoCoverage: 27

Source report: [mutation-testing-report artifact from run 34674810397](https://github.com/microsoft/testfx/actions/runs/34674810397). No newer report is available yet - the 2026-09-13, 2026-09-14, and 2026-09-15 runs all failed before producing one.

## Known Equivalent Mutants

*(None recorded yet. Candidates considered but not yet classified as equivalent - kept as open hotspots above pending deeper investigation: the `RegisterClientSerializers` double-checked-locking idempotency mutants (`SerializerUtilities.ClientSerializers.cs`) may be partially equivalent under single-threaded test execution but were not investigated deeply enough to record with confidence.)*

### 2026-09-15 05:09 UTC - [Run](https://github.com/microsoft/testfx/actions/runs/34931604144)
- ⚠️ Run failed/cancelled - no report available

### 2026-09-14 05:15 UTC - [Run](https://github.com/microsoft/testfx/actions/runs/34808600446)
- ⚠️ Run failed/cancelled - no report available

### 2026-09-13 05:18 UTC - [Run](https://github.com/microsoft/testfx/actions/runs/34739561186)
- ⚠️ Run failed/cancelled - no report available

### 2026-09-12 06:46 UTC - [Run](https://github.com/microsoft/testfx/actions/runs/34674810397)
- Score: 37.35% (Killed 140 / Survived 166 / No Coverage 141 / Timeout 43)
- 🔧 Opened PR (creation queued): kills mutant in `Client/MtpServerConnector.cs:253` (`WaitBoundedAsync` task.IsCompleted boolean mutation) - branch `mutation-test-improver/wait-bounded-async-task-completed`. Verified locally: new tests pass against unmutated code, and a full local Stryker re-run confirmed mutant id 401 flipped from `Survived` to `Killed` (project score after fix: 37.55%, Killed 142 / Survived 164).
- 🟰 No other fix attempted this run: verification budget (2 total Stryker runs) was spent on this candidate; `MtpServerConnector.cs` timeout-clamping equality mutants (lines 256, 261) remained survived and are carried to Suggested Actions above.

> 🤖 **Automated content by GitHub Copilot.** Generated by the [Mutation Test Improver](https://github.com/microsoft/testfx/actions/runs/34932046279/agentic_workflow) workflow. · copilot · auto · 35.8 AIC · ⌖ 12.9 AIC · ⊞ 15.7K · [◷]( · [◷](https://github.com/search?q=repo%3Amicrosoft%2Ftestfx+is%3Aissue+%22gh-aw-workflow-call-id%3A+microsoft%2Ftestfx%2Fmutation-test-improver%22&type=issues))

### PR DESCRIPTION
**Pull request by Evangelink**
The September mutation report identified untested ServerMode client behavior, especially in client-specific JSON-RPC deserialization and connection failure paths. This adds focused coverage for observable behavior instead of attempting to test the two `WaitBoundedAsync` boundary mutants, which are behaviorally equivalent.

The tests now cover:
- client JSON-RPC request, notification, response, invalid ID, invalid header, and malformed-shape handling;
- artifact and run-response defaults when optional data is absent;
- JSON-RPC server error propagation, including the error code and message;
- disposal while a request is pending;
- server-request handler failures returning a null response and emitting a warning.

The fake server forwards the configured client logger and can send an error response, providing the smallest seams needed for these assertions.

Validation: the targeted project builds with zero warnings and errors, and the ServerMode client unit suite passes on both `net8.0` and `net462`.

Fixes: #11212

### CODE REVIEW

File: test/UnitTests/Microsoft.Testing.Platform.ServerMode.Client.Sources.UnitTests/MtpServerClientTests.cs [hunk 1]
```diff
@@ -659,6 +855,26 @@ public async Task ServerInitiatedRequest_WithHandler_InvokesHandler()
...
+        });
+        _ = await WithTimeoutAsync(client.InitializeAsync(TestContext.CancellationToken)).ConfigureAwait(false);
+        client.ServerRequestHandler = (_, _, _) => throw new InvalidOperationException("Debugger launch failed.");
+
+        ResponseMessage response = await WithTimeoutAsync(
+            server.SendServerRequestAsync(ClientAttachDebuggerMethod)).ConfigureAwait(false);
+
+        Assert.IsNull(response.Result);
+        Assert.Contains(ClientAttachDebuggerMethod, log.ToString());
+        Assert.Contains("Debugger launch failed.", log.ToString());
```
[the remark this answers is not in the document]
Comment (Evangelink) [reply test/UnitTests/Microsoft.Testing.Platform.ServerMode.Client.Sources.UnitTests/MtpServerClientTests.cs]: Updated the test to capture structured `(Level, Message)` entries and assert that the matching handler-failure entry is emitted at `MtpClientLogLevel.Warning`. The message content assertions remain part of the same predicate.

<sub>Auto-replied by the [GitHub Copilot app](https://gh.io/app)</sub>

**Review by 0101 — APPROVED** [review]
